docs/tasks/setup_all_or_nothing.md

```diff
@@ -0,0 +1,456 @@
+job1-0-9pvs8    1/1     Running     0          28m
+job1-1-w9zht    1/1     Running     0          28m
+job1-10-fg99v   1/1     Running     0          28m
+job1-11-4gspm   1/1     Running     0          28m
+job1-12-w5jft   1/1     Running     0          28m
+job1-13-8d5jk   1/1     Running     0          28m
+job1-14-h5q8x   1/1     Running     0          28m
+job1-15-kkv4j   0/1     Pending     0          28m
+job1-16-frs8k   0/1     Pending     0          28m
+job1-17-g78g8   0/1     Pending     0          28m
+job1-18-2ghmt   0/1     Pending     0          28m
+job1-19-4w2j5   0/1     Pending     0          28m
+job1-2-9s486    1/1     Running     0          28m
+job1-3-s9kh4    1/1     Running     0          28m
+job1-4-52mj9    1/1     Running     0          28m
+job1-5-bpjv5    1/1     Running     0          28m
+job1-6-7f7tj    1/1     Running     0          28m
+job1-7-pnq7w    1/1     Running     0          28m
+job1-8-7s894    1/1     Running     0          28m
+job1-9-kz4gt    1/1     Running     0          28m
+job2-0-x6xvg    1/1     Running     0          28m
+job2-1-flkpj    1/1     Running     0          28m
+job2-10-vf4j9   1/1     Running     0          28m
+job2-11-ktbld   0/1     Pending     0          28m
+job2-12-sf4xb   1/1     Running     0          28m
+job2-13-9j7lp   0/1     Pending     0          28m
+job2-14-czc6l   1/1     Running     0          28m
+job2-15-m77zt   0/1     Pending     0          28m
+job2-16-7p7fs   0/1     Pending     0          28m
+job2-17-sfdmj   0/1     Pending     0          28m
+job2-18-cs4lg   0/1     Pending     0          28m
+job2-19-x66dt   0/1     Pending     0          28m
+job2-2-hnqjv    1/1     Running     0          28m
+job2-3-pkwhw    1/1     Running     0          28m
+job2-4-gdtsh    1/1     Running     0          28m
+job2-5-6swdc    1/1     Running     0          28m
+job2-6-qb6sp    1/1     Running     0          28m
+job2-7-grcg4    0/1     Pending     0          28m
+job2-8-kg568    1/1     Running     0          28m
+job2-9-hvwj8    0/1     Pending     0          28m
+```
+
+These jobs are now deadlock'ed and are not going to be able to make progress.
+
+#### Cleanup
+
+Clean up the jobs by:
+```shell
+kubectl delete -f quick-job.yaml
+kubectl delete -f /tmp/job1.yaml
+kubectl delete -f /tmp/job2.yaml
+```
+
+### 3. Run when wait for pods ready is enabled
+
+#### Enable waitForPodsReady
+
+Update the Kueue configuration following the instructions [here](#enabling-waitforpodsready).
+
+#### Run the jobs
+
+Run the `start.sh` script
+
+```yaml
+sed 's/_ID_/1/g' job-template.yaml > /tmp/job1.yaml
+sed 's/_ID_/2/g' job-template.yaml > /tmp/job2.yaml
+kubectl create -f quick-job.yaml
+kubectl create -f /tmp/job1.yaml
+kubectl create -f /tmp/job2.yaml
+```
+
+#### Monitor the progress
+
+Execute the following command in a couple of seconds internals to monitor
+the progress:
+```shell
+kubectl get pods
+```
+
+We omit the pods of the completed `quick` job for brevity.
+
```

Comment:
Do you think we should also display the job status like:
```
kubectl get job -o custom-columns="NAME":.metadata.name,"COMPLETIONS":.spec.completions,"SUSPEND":.spec.suspend

NAME         COMPLETIONS   SUSPEND
sample-job   3             true
```

Reply:
I'm not convinced it will give us much. The fact that `job2` remains suspended could be assumed due to the fact that it does not have any pods running in the first output for `kubectl got pods`. 

Then, using two commands could make the timing important if someone wants to do the experiment on their own for the first time. If we want to go for multiple monitoring commands, I think we would need to provide commands for events monitoring (workload and job), which is a complication I wanted to avoid, if possible, for now. 

Also, in the command you provide the `.spec.completions` does not change, so I see no point including. Instead we could monitor `.status.succeeded` and `.status.ready`, but it does not seem necessary. WDYT?